app/protocols/ssb-protocol.js

```diff
@@ -3,9 +3,20 @@ const fetchToHandler = require('./fetch-to-handler')
 module.exports = async function createHandler (options, session) {
   return fetchToHandler(async () => {
     const { makeSsbFetch } = require('ssb-fetch')
+    const appname = process.env.ssb_appname || options.appname || 'ssb'
 
-    const fetch = makeSsbFetch(options)
+    /** connect to running ssb-server */
+    if (!!options.ssbd === false) {
```

Comment:
Would it make sense to try to do an HTTP request to a local sbot server to see if it exists before initializing ssbd?

Reply:
Not all sbot's have an http server. They're muxrpc based. Some do have an http server if they're running `ssb-ws` or have added a server, but it's not required. That's if I'm understanding your request correctly :)

[`ssb-party`](https://www.npmjs.com/package/ssb-party) has the functionality you want but hasn't been maintained for quite some time I think. It's certainly not a big deal to implement this and I had considered it, but I find myself that I sometimes have a couple of different sbots running at the same time on different ports and it's super easy to connect to the one you want by setting the `ssb_appname` env var to the server you want.  

That functionality is already implemented in `ssb-fetch`. The way it works with this pull request is:

* default config baked into agregore - `~/.agregore-ssb`
* user config that overrides in `~/.agregorerc`
* env var that overrides the others e.g. `export ssb_appname=ssb`

So if you're a new member of the scuttleverse you can just use agregore to access the network without having to set anything.

If you're an existing user and have a log in `~/.ssb` you can set `appname` to `ssb` to use that identity. You can either use the bundled `ssbd` server or set `ssbd.runServer` to false to connect to a running sbot.

And then if you're hacking on ssb and running more than one sbot it's easy to manage with `ssb_appname` env var.

Different configurations are possible by twiddling the env var and `.agregorerc` for flexibility and it all works as you'd expect (much better than my earlier attempt with shipyard!). And in the simple case you don't have to configure anything. Just run agregore.